lib/middleware/template.js

```diff
@@ -40,10 +40,22 @@ module.exports = async (ctx, next) => {
         }
 
         if (ctx.state.data) {
+            if (ctx.state.data.title) {
+                ctx.state.data.title = ctx.state.data.title.replace(/\s+/g, ' ').trim();
+            }
+
+            if (ctx.state.data.subtitle) {
+                ctx.state.data.subtitle = ctx.state.data.subtitle.replace(/\s+/g, ' ').trim();
+            }
+
+            if (ctx.state.data.author) {
+                ctx.state.data.author = ctx.state.data.author.replace(/\s+/g, ' ').trim();
+            }
+
             ctx.state.data.item &&
                 ctx.state.data.item.forEach((item) => {
                     if (item.title) {
```

Comment:
```suggestion
                    if (item.title && rWhiteSpace.test(item.title)) {
```

Reply:
We should not do this, since a title trimming must happen even if there is no whitespace to be collapsed.